Tests/Runtime/Shape/BezierMeshTests.cs

```diff
@@ -0,0 +1,188 @@
+#if USING_SPLINES
+
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+using Unity.Mathematics;
+using UnityEngine;
+using UnityEngine.ProBuilder;
+using UnityEngine.Splines;
+using Object = UnityEngine.Object;
+using Spline = UnityEngine.Splines.Spline;
+
+class BezierMeshTests
+{
+    BezierMesh m_Mesh;
+
+    [SetUp]
+    public void SetUp()
+    {
+        GameObject go = new GameObject();
+        m_Mesh = go.AddComponent<BezierMesh>();
+        float3 tan = new float3(0f, 0f, 2f);
+        float3 p1 = new float3(3f, 0f, 0f);
+
+        var spline = new List<Spline> { new Spline() };
+        spline[0].Add(new BezierKnot(float3.zero, -tan, tan, Quaternion.identity));
+        spline[0].Add(new BezierKnot(p1, p1 + tan, p1 + -tan, Quaternion.identity));
+
+        Assume.That(m_Mesh.splineContainer, Is.Not.Null);
+
+        m_Mesh.splineContainer.Splines = spline;
+        m_Mesh.ExtrudeMesh();
+    }
+
+    [TearDown]
+    public void TearDown()
+    {
+        Object.DestroyImmediate(m_Mesh);
+    }
+
+    [Test]
+    public void BezierMesh_AddingAKnot_ExtrudesMesh()
+    {
+        Assert.That(m_Mesh, Is.Not.Null);
+
+        var newKnot = new BezierKnot(new float3(6f, 0f, 0f));
+        m_Mesh.splineContainer.Splines[0].Add(newKnot);
+        var vertices = m_Mesh.mesh.GetVertices();
+
+        var isVertexAtExpectedPosition = false;
+
+        // If a knot is added at x = 6f, and the mesh is extruded, then we expect to have at least one vertex at that position
+        foreach (var vertex in vertices)
+        {
+            if (Mathf.Approximately(vertex.position.x, 6f))
+            {
+                isVertexAtExpectedPosition = true;
```

Comment:
depending on your settings for the bezier mesh extrusion you can also count the number of vertices with a position.x at 6 and check that the count is right at the end, that might be better this way.